streams/src/main/java/org/apache/kafka/streams/kstream/KTable.java

```diff
@@ -17,378 +17,831 @@
+     * {@link  StreamsConfig} via parameter {@link StreamsConfig#APPLICATION_ID_CONFIG APPLICATION_ID_CONFIG}, "XXX" is
+     * an internal generated name, and "-repartition" is a fixed suffix.
+     * You can retrieve all generated internal topic names via {@link KafkaStreams#toString()}.
+     * <p>
+     * All data of this {@link KTable} will be redistributed through the repartitioning topic by writing all update
+     * records to and rereading all update records from it, such that the resulting {@link KGroupedTable} is partitioned
+     * on the new key.
+     *
+     * @param selector   a {@link KeyValueMapper} that computes a new grouping key and value to be aggregated
+     * @param keySerde   key serdes for materializing this stream,
+     *                   if not specified the default serdes defined in the configs will be used
+     * @param valueSerde value serdes for materializing this stream,
+     *                   if not specified the default serdes defined in the configs will be used
+     * @param <KR>       the key type of the result {@link KGroupedTable}
+     * @param <VR>       the value type of the result {@link KGroupedTable}
+     * @return a {@link KGroupedTable} that contains the re-grouped records of the original {@link KTable}
      */
-    <V1, R> KTable<K, R> leftJoin(KTable<K, V1> other, ValueJoiner<? super V, ? super V1, ? extends R> joiner);
-
+    <KR, VR> KGroupedTable<KR, VR> groupBy(final KeyValueMapper<? super K, ? super V, KeyValue<KR, VR>> selector,
+                                           final Serde<KR> keySerde,
+                                           final Serde<VR> valueSerde);
 
     /**
-     * Group the records of this {@link KTable} using the provided {@link KeyValueMapper}.
-     * 
-     * @param selector      select the grouping key and value to be aggregated
-     * @param keySerde      key serdes for materializing this stream,
-     *                      if not specified the default serdes defined in the configs will be used
-     * @param valueSerde    value serdes for materializing this stream,
-     *                      if not specified the default serdes defined in the configs will be used
-     * @param <K1>          the key type of the {@link KGroupedTable}
-     * @param <V1>          the value type of the {@link KGroupedTable}
-     *
-     * @return a {@link KGroupedTable} that contains the re-partitioned records of this {@link KTable}
+     * Join records of this {@link KTable} with another {@link KTable}'s records using non-windowed inner equi join.
+     * The join is a primary key join with join attribute {@code thisKTable.key == otherKTable.key}.
+     * The result is an ever updating {@link KTable} that represents the <em>current</em> (i.e., processing time) result
+     * of the join.
+     * <p>
+     * The join is computed by (1) updating the internal state if one {@link KTable} and (2) performing a lookup for a
```

Comment:
Why?
To my knowledge, for two things that are listed there is no comma: "A and B"
Only for three or more, there are command: "A, B, C, and D"

Correct my if I am wrong.